pkg/expr/mathexp/reduce.go

```diff
@@ -69,6 +69,17 @@ func Count(fv *Float64Field) *float64 {
 	return &f
 }
 
+func Last(fv *Float64Field) *float64 {
+	var f float64
+	if fv.Len() == 0 {
+		nan := math.NaN()
```

Comment:
No need to change this, but since `f` is declared and we return in the if statement then we can assign `math.NaN()` to `f`:

```suggestion
		f = math.NaN()
		return &f
```